regression/cbmc-incr-smt2/nondeterministic-int-assert/incremental_solver_called.desc

```diff
@@ -0,0 +1,12 @@
+CORE
+empty.c
+--incremental-smt2-solver z3
+WARNING: Leaving variables of static lifetime uninitialised.
+Passing problem to incremental SMT2 solving via "z3"
```

Comment:
Should we check the passing/printing/etc of this string (how does it handle spaces or special characters in this string)?

Reply:
There is no special handling here because this is a rough initial version. Once we have more of the sub-process communication stuff in place we can can test this with examples that have spaces or other interesting characters. The reason to avoid over-thinking this now is in case we change the internal representation with things like splitting into multiple strings, which could entail rework of the printing if we are overly thorough about this upfront. Unless you know of a handy string sanitisation / escaping function in the code base already, which we could just slot in here...

Reply:
`util/run.cpp` has some shell quoting stuff we could reuse?

That said I agree we don't need to commit too seriously here. I was mostly thinking the test could check that if we pass in something like `"z3 -T:600"` (to have a 600 second timeout) that this is handled fine for now and shows up in the output.

Reply:
This seems like a finer point that should be looked at as part of the follow-up sub-process communication work.